lib/rucio/core/did_meta_plugins/elasticsearch_meta.py

```diff
@@ -0,0 +1,406 @@
+        client_key = client_key or config.config_get('metadata', 'client_key', False, None)
+
+        self.es_config = {
+            'hosts': hosts,
+            'timeout': request_timeout,
+            'max_retries': max_retries,
+            'retry_on_timeout': retry_on_timeout
+        }
+        if user and password:
+            self.es_config['basic_auth'] = (user, password)
+
+        if use_ssl:
+            self.es_config.update({
+                'ca_certs': ca_certs,
+                'verify_certs': verify_certs,
+            })
+            if client_cert and client_key:
+                self.es_config.update({
+                    'client_cert': client_cert,
+                    'client_key': client_key
+                })
+
+        self.client = Elasticsearch(**self.es_config)
+        self.plugin_name = "ELASTIC"
+
+    def drop_index(self) -> None:
+        self.client.indices.delete(index=self.index)
+
+    def get_metadata(
+        self,
+        scope: "InternalScope",
+        name: str,
+        *,
+        session: "Optional[Session]" = None
+    ) -> dict[str, Any]:
+        """
+        Get data identifier metadata.
+
+        :param scope: The scope name
+        :param name: The data identifier name
+        :param session: The database session in use
+        :returns: The metadata for the did
+        :raises DataIdentifierNotFound: If the DID metadata is not found.
+        :raises RucioException: If another error occurs during the process.
+        """
+
+        doc_id = f"{scope.internal}{name}"
+        try:
+            doc = self.client.get(index=self.index, id=doc_id)["_source"]
+        except elastic_exceptions.NotFoundError as err:
+            raise exception.DataIdentifierNotFound(f"No metadata found for DID '{scope}:{name}' not found") from err
+        except Exception as err:
+            raise exception.RucioException(err)
+        return doc
+
+    def set_metadata(
+        self,
+        scope: "InternalScope",
+        name: str,
+        key: str,
+        value: str,
+        recursive: bool = False,
+        *,
+        session: "Optional[Session]" = None
+    ) -> None:
+        """
+        Set single metadata key.
+
+        :param scope: the scope of did
+        :param name: the name of the did
+        :param key: the key to be added
+        :param value: the value of the key to be added
+        :param recursive: recurse into DIDs (not supported)
+        :param session: The database session in use
+        :raises DataIdentifierNotFound: If the DID is not found.
+        :raises RucioException: If an error occurs while setting the metadata.
+        """
+        self.set_metadata_bulk(scope=scope, name=name, meta={key: value}, recursive=recursive, session=session)
+
+    def set_metadata_bulk(
+        self,
+        scope: "InternalScope",
+        name: str,
+        meta: dict[str, Any],
+        recursive: bool = False,
```

Comment:
Could you add a `raise exception.UnsupportedOperation(f"'{self.plugin_name.lower()}' metadata module does not currently support recursive searches")` if `recursive = True`, in the same way as you've done with `list_dids`?